modules/programs/bat.nix

```
@@ -159,10 +159,15 @@ in {
         };
       })));
 
+      # NOTE: run `bat cache --build` in a fresh temporary directory to work around failure when ~/cache exists
+      # https://github.com/sharkdp/bat/issues/1726
       home.activation.batCache = hm.dag.entryAfter [ "linkGeneration" ] ''
         (
           export XDG_CACHE_HOME=${escapeShellArg config.xdg.cacheHome}
           verboseEcho "Rebuilding bat theme cache"
+          tempdir=$(mktemp -d)
+          trap "rm -r '$tempdir'" EXIT
+          cd "$tempdir"
```

I think you could simplify this to
```suggestion
          cd "${pkgs.emptyDirectory}"
```

---

That's a nice solution, thanks!